src/search/SearchMessagesScreen.js

```diff
@@ -62,11 +70,17 @@ class SearchMessagesScreenInner extends PureComponent<Props, State> {
    * Stores the fetched messages in the Redux store. Does not read any
    * of the component's data except `props.dispatch`.
    */
-  fetchSearchMessages = async (query: string): Promise<$ReadOnlyArray<Message>> => {
+  fetchSearchMessages = async (
+    query: string,
+  ): Promise<{
+    foundNewest: boolean,
+    foundOldest: boolean,
+    messages: $ReadOnlyArray<Message>,
+  }> => {
     const fetchArgs = {
       narrow: SEARCH_NARROW(query),
       anchor: LAST_MESSAGE_ANCHOR,
-      numBefore: 20,
+      numBefore: config.messagesPerRequest,
```

Comment:
Let's do this as its own little commit before the main one. It's an important and useful user-visible change on its own, logically independent of the main change, and it deserves its own commit and commit message.

I think basically the point here is that now that we're no longer trying to do a search on every keystroke, there's no need to be hyper-frugal on how much we fetch up front. By the time we actually do a search request, the user has said this is a search they actually want the results for, just like they might by navigating to some other narrow like a conversation. So we should fetch the same number of messages as for other narrows — enough so that they're not just going to immediately want more.